spec/ndx-extracellular-channels.extensions.yaml

```diff
@@ -0,0 +1,287 @@
+    - - num_points
+      - x, y
+    - - num_points
+      - x, y, z
+    shape:
+    - - null
+      - 2
+    - - null
+      - 3
+    doc: The coordinates of the nodes of the polygon that describe the shape (contour)
+      of the probe, in micrometers. The first and last points are connected to close
+      the polygon. e.g., [(-20., -30.), (20., -110.), (60., -30.), (60., 190.), (-20.,
+      190.)].See 'probe_planar_contour' in https://probeinterface.readthedocs.io/en/main/format_spec.html
+      for more details.
+  groups:
+  - name: contacts_table
+    neurodata_type_inc: ContactsTable
+    doc: Neural probe contacts, compatible with the ProbeInterface specification
+- neurodata_type_def: Probe
+  neurodata_type_inc: Device
+  doc: Specific instance of a neural probe object.
+  attributes:
+  - name: identifier
+    dtype: text
+    doc: Identifier of the probe, usually the serial number.
+    required: false
+  links:
+  - name: probe_model
+    target_type: ProbeModel
+    doc: The model of the probe used to record the data.
+- neurodata_type_def: ProbeInsertion
+  neurodata_type_inc: NWBContainer
+  default_name: probe_insertion
+  doc: Metadata about the insertion of a probe into the brain, which can be used to
+    determine the location of the probe in the brain.
+  attributes:
+  - name: reference
+    dtype: text
+    doc: Reference point for `insertion_position_in_mm` coordinates, e.g., "bregma
+      at the cortical surface".
+    required: false
+  - name: hemisphere
+    dtype: text
+    doc: The hemisphere ("left" or "right") of the targeted location of the optogenetic
+      stimulus site. Should be consistent with `insertion_position_in_mm.ml` coordinate
+      (left = ml < 0, right = ml > 0).
+    required: false
+  - name: depth_in_mm
+    dtype: float
+    doc: Depth that the probe was driven along `insertion_angle` starting from `insertion_position_in_mm`,
+      in millimeters.
+    required: false
+- neurodata_type_def: ChannelsTable
+  neurodata_type_inc: DynamicTable
+  default_name: ChannelsTable
+  doc: Metadata about the channels used in an extracellular recording from a single
+    probe.
+  attributes:
+  - name: reference_mode
+    dtype: text
+    doc: The reference mode used for the recording; e.g. 'external wire', 'common
+      reference'.
+    required: false
+  datasets:
+  - name: contact
+    neurodata_type_inc: DynamicTableRegion
+    doc: The row in a ContactsTable that represents the contact used as a channel.
+    quantity: '?'
+  - name: reference_contact
+    neurodata_type_inc: DynamicTableRegion
+    doc: The row in a ContactsTable that represents the contact used as a reference.
+    quantity: '?'
+  - name: filter
+    neurodata_type_inc: VectorData
+    dtype: text
+    doc: The filter used on the raw (wideband) voltage data from this contact, including
+      the filter name and frequency cutoffs, e.g., 'High-pass filter at 300 Hz.'
+    quantity: '?'
+  - name: estimated_position_in_mm
```

Comment:
It looks as if you may have changed various things in the schema since last we met

But the mermaid has not been updated; can you bring the two in sync?

Reply:
Done